### PR TITLE
Add Teager-Kaiser energy

### PR DESCRIPTION
**Pull request by GuillaumeCorda**

### CODE REVIEW

**Comment by GuillaumeCorda** [comment]
Thank you for helping me !

----- Mail original -----
De: "Alexandre Gramfort" <notifications@github.com>
À: "mne-tools/mne-features" <mne-features@noreply.github.com>
Cc: "Guillaume Corda" <guillaume.corda@telecom-paristech.fr>, "Author" <author@noreply.github.com>
Envoyé: Mardi 17 Avril 2018 18:20:41
Objet: Re: [mne-tools/mne-features] Add Teager-Kaiser energy (#28)

ok let's try again with a clean PR history.
Do:

git checkout master
git pull upstream master
git checkout -b taeger-kaiser_energy
# do the changes
git commit -am "commit message"
git push 	GuillaumeCorda taeger-kaiser_energy

you should then have a clean history.

basically the pb was that you started the branch on an old version of the master brain.

**Comment by agramfort** [comment]
you should also use gitk or gitg or tig to view the git history. to know
where you are.

File: mne_features/univariate.py [hunk 1]
```diff
@@ -28,7 +28,43 @@ def get_univariate_funcs(sfreq):
     -------
     univariate_funcs : dict
     """
+<<<<<<< HEAD
```
Comment (jbschiratti) [inline mne_features/univariate.py]: A conflict remains!
You should keep ``return _get_feature_funcs(sfreq, __name__)`` and remove the rest.

**Comment by agramfort** [comment]
never push without running the tests....

**Comment by agramfort** [comment]
I see no update in api.rst so the new function is not documented.

also the tests are really just smoke tests now. You should test that the values
are correct. This is a more general problem of all the test currently. You should
synthesize signals and check that values are correct (using maths results
on function or testing values using assert_almost_equal). Example of tests:
you get positive numbers? feature values are higher for a sinusoid than for
white noise? be creative but you need really unit tests.

cc @jbschiratti

**Comment by jbschiratti** [comment]
@agramfort I agree. For now, most tests only test the shape of the array returned by the function.
For some (most?) function, writing meaningful tests might not be straightforward...
I think an issue should be opened for this. Further discussion on this could go on in the issue.

**Comment by agramfort** [comment]
yes in a next PR
